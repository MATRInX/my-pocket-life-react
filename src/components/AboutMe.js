import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { LINKS } from '../fixtures/links';
import selectAboutMePage from '../selectors/aboutMePage';

export const AboutMe = ({ aboutMePage }) => (
  <div>
    {
      aboutMePage === undefined ? (
        <div>Loading...</div>
      ) : (
        <div>
          <h2>Kilka słów o mnie</h2>
          <Link to={LINKS.ABOUT_ME}>
            <img src={aboutMePage.image}/>
          </Link>
          <p>
            Mam na imię Aneta i od półtora roku jestem mamą. Wraz z mężem i synem mieszkamy w Tarnowskich Górach.
          </p>
          <Link to={LINKS.ABOUT_ME}>Czytaj więcej...</Link>
        </div>
      )
    }    
  </div>
);

const mapStateToProps = ({ pages } = state) => ({
  aboutMePage: selectAboutMePage(pages)
});

export default connect(mapStateToProps)(AboutMe);