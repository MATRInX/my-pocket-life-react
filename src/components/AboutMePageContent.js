import React from 'react';

const AboutMePageContent = () => (
  <div>
    AboutMePageContent content...
  </div>
);

export default AboutMePageContent;