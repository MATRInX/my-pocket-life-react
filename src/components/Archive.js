import React from 'react';
import { connect } from 'react-redux';
import selectArchiveCategory from '../selectors/archiveCategory';
import { setArchiveMonth, setArchiveYear } from '../actions/filters';
import { history } from '../routers/AppRouter';

export class Archive extends React.Component {
  state = {
    selectValue: this.props.selectValue ? this.props.selectValue : 
      this.props.filters.archiveMonth + '|' + this.props.filters.archiveYear
  };

  componentDidMount() {
    if (this.state.selectValue !== '0|0') {
      const filters = this.state.selectValue.split('|');
      this.props.setArchiveYear(Number(filters[1]));
      this.props.setArchiveMonth(Number(filters[0]));
    }
  }

  onSelectChange = e => {
    const selectValue = e.target.value;
    this.setState(() => ({ selectValue }));
    const filters = selectValue.split('|');
    this.props.setArchiveYear(Number(filters[1]));
    this.props.setArchiveMonth(Number(filters[0]));
    history.push(`/archiwum/${filters[1]}/${filters[0]}`);
  }

  render() {
    return (
      <div>
        <p>Archive content: {this.state.selectValue}</p>
        <select 
          value={this.state.selectValue}
          onChange={this.onSelectChange}
        >
          <option value="0|0">Wybierz miesiÄ…c...</option>
          {
            this.props.archiveData.map((archiveItem, index) => {
              return (
                <option 
                  key={index} 
                  value={archiveItem.month+'|'+archiveItem.year}
                >
                  {archiveItem.text}  
                </option>
              )
            })
          }
        </select>
      </div>
    );
  }
};

const mapStateToProps = ({ posts, filters } = state) => ({
  archiveData: selectArchiveCategory(posts),
  filters
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  setArchiveMonth: month => dispatch(setArchiveMonth(month)),
  setArchiveYear: year => dispatch(setArchiveYear(year))
});

export default connect(mapStateToProps, mapDispatchToProps)(Archive);