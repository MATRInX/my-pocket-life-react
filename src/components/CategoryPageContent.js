import React from 'react';

const CategoryPageContent = (props) => (
  <div>
    CategoryPageContent - category:
    <p>{props.match.params.category}</p>
  </div>
);

export default CategoryPageContent;