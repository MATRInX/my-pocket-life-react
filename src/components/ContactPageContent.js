import React from 'react';

const ContactPageContent = () => (
  <div>
    ContactPageContent content...
  </div>
);

export default ContactPageContent;