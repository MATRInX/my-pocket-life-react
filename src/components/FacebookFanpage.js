import React from 'react';

const FacebookFanpage = () => (
  <div>FacebookFanpage content</div>
);

export default FacebookFanpage;