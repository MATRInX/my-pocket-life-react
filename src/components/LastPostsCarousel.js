import React from 'react';
import { connect } from 'react-redux';
import { Carousel } from 'react-responsive-carousel';
import selectLastPostsCarousel from '../selectors/lastPostsCarousel';

const temporaryStyle = {
  width: '400px'
};

const pStyle = {
  background: 'transparent',
  color: 'black',
  fontWeight: '700',
  textTransform: 'uppercase'
};

export const LastPostsCarousel = ({ carouselPosts } = props) => (
  <div style={temporaryStyle}>
    {
      carouselPosts.length === 0 ? (
        <p>Loading...</p>
      ) : (
        <Carousel 
          autoPlay={true} 
          emulateTouch={true} 
          infiniteLoop={true} 
          interval={5000}
          showIndicators={false}
          showStatus={false} 
          showThumbs={false} 
        >
          {
            carouselPosts.map((singlePost, index) => (
              <div key={index}>
                <img src={singlePost.image} width="300" height="300"/>
                <p className="legend" style={pStyle}>{singlePost.title}</p>
              </div>
            ))
          }      
        </Carousel>
      )
    }    
  </div>
);

const mapStateToProps = ({ posts } = state) => ({
  carouselPosts: selectLastPostsCarousel(posts)
});

export default connect(mapStateToProps)(LastPostsCarousel);