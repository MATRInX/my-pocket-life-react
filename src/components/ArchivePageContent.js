import React from 'react';
import Archive from './Archive';
import PostList from './PostsList';

const ArchivePageContent = ({ match }) => (
  <div>
    ArchivePageContent content...
    <PostList />
    <Archive selectValue={match.params.month +'|'+match.params.year}/>
  </div>
);

export default ArchivePageContent;