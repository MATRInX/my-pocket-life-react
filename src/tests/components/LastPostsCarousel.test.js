import React from 'react';
import { shallow } from 'enzyme';
import { LastPostsCarousel } from '../../components/LastPostsCarousel';
import { dummyPosts as carouselPosts } from '../../fixtures/dummyData';

test('should correctly render LastPostsCarousel component', () => {
  const wrapper = shallow(<LastPostsCarousel carouselPosts={carouselPosts}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render loading information when there is no last posts', () => {
  const wrapper = shallow(<LastPostsCarousel carouselPosts={[]}/>);
  expect(wrapper).toMatchSnapshot();
});