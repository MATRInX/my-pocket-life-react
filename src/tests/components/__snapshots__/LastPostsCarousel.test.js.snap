// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LastPostsCarousel component 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <Carousel
    autoPlay={true}
    axis="horizontal"
    centerMode={false}
    centerSlidePercentage={80}
    dynamicHeight={false}
    emulateTouch={true}
    infiniteLoop={true}
    interval={5000}
    onChange={[Function]}
    onClickItem={[Function]}
    onClickThumb={[Function]}
    selectedItem={0}
    showArrows={true}
    showIndicators={false}
    showStatus={false}
    showThumbs={false}
    statusFormatter={[Function]}
    stopOnHover={true}
    swipeScrollTolerance={5}
    swipeable={true}
    transitionTime={350}
    useKeyboardArrows={false}
    verticalSwipe="standard"
    width="100%"
  >
    <div
      key="0"
    >
      <img
        height="300"
        width="300"
      />
      <p
        className="legend"
        style={
          Object {
            "background": "transparent",
            "color": "black",
            "fontWeight": "700",
            "textTransform": "uppercase",
          }
        }
      >
        Sample title1
      </p>
    </div>
    <div
      key="1"
    >
      <img
        height="300"
        width="300"
      />
      <p
        className="legend"
        style={
          Object {
            "background": "transparent",
            "color": "black",
            "fontWeight": "700",
            "textTransform": "uppercase",
          }
        }
      >
        Sample title2
      </p>
    </div>
    <div
      key="2"
    >
      <img
        height="300"
        width="300"
      />
      <p
        className="legend"
        style={
          Object {
            "background": "transparent",
            "color": "black",
            "fontWeight": "700",
            "textTransform": "uppercase",
          }
        }
      >
        Sample title1
      </p>
    </div>
  </Carousel>
</div>
`;

exports[`should render loading information when there is no last posts 1`] = `
<div
  style={
    Object {
      "width": "400px",
    }
  }
>
  <p>
    Loading...
  </p>
</div>
`;
